### PR TITLE
Add "Per.11: Move computation from run time to compile time"

### PR DESCRIPTION
**Pull request by scraimer**

### CODE REVIEW

**Comment by scraimer** [comment]
Related issue: #936

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -12839,7 +12839,104 @@ Type violations, weak types (e.g. `void*`s), and low level code (e.g., manipulat
...
+
+Here's a `constexpr` function for calculating Fibonacci Numbers:
+
+    static constexpr int fib(int const n)
+    {
+        return (n <= 0) ? 0 :
+                  (n == 1) ? 1 :
+                     fib(n-1) + fib(n-2);
+    }
+
```
Comment (MikeGitb) [inline CppCoreGuidelines.md]: I don't think it is a good idea to show recursive Fibonacci over and over again. If you repeat it often enough, people might start thinking it is actually a good way to do it like this

Comment (MikeGitb) [inline CppCoreGuidelines.md]: Thanks for the change, but what is the purpose of `ith_minus_one`? And you are missing a return.

Comment (MikeGitb) [inline CppCoreGuidelines.md]: May I suggest something like:

	constexpr int fib(int const n)
	{
		int f_n = 0, f_next = 1;
		for (int i = 0; i < n; ++i)
		{
			auto t = f_next;
			f_next = f_next + f_n;
			f_n = t;
		}
		return f_n;
	}

Comment (scraimer) [inline CppCoreGuidelines.md]: Thank you for taking the time to review this! I really appreciate it.

**Comment by AndrewPardoe** [comment]
Thank you for the suggested rule! Per our editors' discussion we have some feedback.

First, don't inline all the other fixes that have happened into your PR. Just add this rule, Git will merge it nicely.

Second, while we appreciate the suggested rule it is a little lengthy. Maybe you could start with an Issue where we discuss what points should be made and we can sketch out the rule we'd like to see in that discussion? Then you can write a PR that fits with the style of the section we can merge more easily.

**Comment by scraimer** [comment]
Sounds like a plan! Thank you for taking the time to look it over.

I had originally opened #940 to discuss the suggested content of the rule. Perhaps it was too bold to open the PR so quickly.  Could you point me to an example issue which I should emulate?  (I'm looking at the closed issues for the past 3 months, and with only a couple exceptions, they all end with just a few comments).

**Pull request closed without merging**